tripal_bulk_loader/includes/tripal_bulk_loader.loader.inc

```diff
@@ -209,12 +209,12 @@ function tripal_bulk_loader_load_data($nid, $job_id) {
     foreach ($record_array['fields'] as $field_index => $field_array) {
 
       $default_data[$priority]['table'] = $record_array['table'];
-      $default_data[$priority]['mode'] = ($record_array['mode']) ? $record_array['mode'] : 'insert';
-      $default_data[$priority]['select_if_duplicate'] = ($record_array['select_if_duplicate']) ? $record_array['select_if_duplicate'] : 0;
-      $default_data[$priority]['update_if_duplicate'] = ($record_array['update_if_duplicate']) ? $record_array['update_if_duplicate'] : 0;
-      $default_data[$priority]['disabled'] = ($record_array['disable']) ? $record_array['disable'] : 0;
-      $default_data[$priority]['optional'] = ($record_array['optional']) ? $record_array['optional'] : 0;
-      $default_data[$priority]['select_optional'] = ($record_array['select_optional']) ? $record_array['select_optional'] : 0;
+      $default_data[$priority]['mode'] = empty($record_array['mode']) ? 'insert' : $record_array['mode'];
```

Comment:
I think we want to use array_key_exists here instead of empty. This is closer to what I originally meant when I wrote this code.

```suggestion
      $default_data[$priority]['mode'] = array_key_exists('mode', $record_array) ? $record_array['mode'] : 'insert';
```

Reply:
Okay, suggested change is made. I used empty() because it seemed more compact, but this is perfect.

Reply:
I tested again with this version, worked as expected, no errors.